apic_ml2/neutron/plugins/ml2/drivers/cisco/apic/mechanism_apic.py

```diff
@@ -908,7 +908,8 @@ def _check_gw_port_operation(self, context, port):
                 context._plugin_context,
                 filters={'id': [p['device_id'] for p in router_gw_ports]})
             routers = [r for r in routers
-                       if r['tenant_id'] != router['tenant_id']]
+                       if r['tenant_id'] != '' and
```

Comment:
this could be written as:
if r['tenant_id'] and r['tenant_id'] != router['tenant_id']